back/bots/models/chat.py

```
@@ -63,12 +82,85 @@ def get_response(self, ai=None):
         
         if self.user.user_account.over_limit():
             return "You have exceeded your daily limit. Please try again tomorrow or upgrade your subscription."
+        
+        if self.bot and self.bot.enable_web_search and settings.TAVILY_API_KEY:
+            tavily_client = TavilyClient(api_key=settings.TAVILY_API_KEY)
+            
+            def web_search(query):
+                try:
+                    results = tavily_client.search(query=query)
+                    return {"results": results.get('results', []), "error": None}
+                except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)